python/graphstorm/eval/evaluator.py

```diff
@@ -341,6 +341,14 @@ def do_eval(self, total_iters, epoch_end=False):
             return True
         return False
 
+    def _get_early_stop_score(self, val_score):
+        """ By default, always get the first validation
+        score as the indicator for early stop.
+        """
+        # First metric is the key eval metric
+        metric = self.metric_list[0]
```

Comment:
Is there a guarantee metric_list will have at least one element?

Reply:
+1. And can we check if the `metric` is in the `val_score`?

Reply:
There is at least one element in the `metric_list`.
But we can add an assert to check ` if the metric is in the val_score`